### DISCUSSION TITLE
Is there a way to deny access to docs by ip

### DISCUSSION THREAD

**Discussion by Actticus** [fastapi/fastapi#5248] [Questions] [answered]
For security measures, I want to give you access to the API by IP documentation. Is there any way to do that?

**Comment by yezz123** [comment]
That Depend on what platform you are using like AWS you can check Docs https://aws.amazon.com/premiumsupport/knowledge-center/api-gateway-resource-policy-access/

**Reply by Actticus** [reply]
No, I mean to give ip access only for one page (docs), other pages will be public

**Comment by davidharting** [comment]
If you have nginx or another reverse proxy sitting in front of Fast API, you could configure such a thing using that. 

Maybe this could also be done with a middleware! That would actually probably be what I would try first.

**Comment by YuriiMotov** [comment, marked as answer]
In addition to [approach with proxy](https://github.com/fastapi/fastapi/discussions/5248#discussioncomment-3389463)

```python
from fastapi import Depends, FastAPI, HTTPException, Request
from fastapi.openapi.docs import get_swagger_ui_html

WHITE_LIST_IPS = {
    "127.0.0.1",
}

async def withe_list_ip(request: Request):
    if request.client.host not in WHITE_LIST_IPS:
        raise HTTPException(status_code=403, detail="Access denied")

app = FastAPI(docs_url=None, redoc_url=None)

@app.get("/docs", include_in_schema=False, dependencies=[Depends(withe_list_ip)])
async def custom_swagger_ui_html():
    return get_swagger_ui_html(
        openapi_url=app.openapi_url,
        title=app.title + " - Swagger UI",
        oauth2_redirect_url=app.swagger_ui_oauth2_redirect_url,
    )
```
